staging/src/k8s.io/code-generator/cmd/client-gen/generators/fake/generator_fake_for_group.go

```diff
@@ -56,7 +62,10 @@ func (g *genFakeForGroup) Namers(c *generator.Context) namer.NameSystems {
 }
 
 func (g *genFakeForGroup) Imports(c *generator.Context) (imports []string) {
-	imports = append(g.imports.ImportLines(), strings.ToLower(fmt.Sprintf("%s \"%s\"", filepath.Base(g.realClientPackage), g.realClientPackage)))
+	imports = g.imports.ImportLines()
+	if len(g.types) != 0 {
+		imports = append(imports, strings.ToLower(fmt.Sprintf("%s \"%s\"", filepath.Base(g.realClientPackage), g.realClientPackage)))
```

Comment:
@sttts these three lines are new. This fix the fake client in case there is no types requires clients in a group. In this case, the imported package is not referenced in the generated fake client, so there will be a compile error.

See the generated diff for the new event API:
https://github.com/caesarxuchao/kubernetes/commit/c984554789f0f05578880bf8659d667e218f28f9#diff-28a38b2d953d816541c1c2508844e543L22

Reply:
We define too many explicit imports and should use the import tracker for that instead.